app/indexing/bm25_index.py

```diff
@@ -49,6 +49,8 @@ def update(self):
         with Session() as session:
             all_paragraphs = session.query(Paragraph).all()
             corpus = [nltk.word_tokenize(_add_metadata_for_indexing(paragraph)) for paragraph in all_paragraphs]
+            if len(corpus) == 0:
```

Comment:
By PEP8 https://peps.python.org/pep-0008/
For sequences, (strings, lists, tuples), use the fact that empty sequences are false:
```python
# Correct:
if not seq:
if seq:
```
```python
# Wrong:
if len(seq):
if not len(seq):
```